Core/ELF/ParamSFO.cpp

```diff
@@ -276,3 +280,34 @@ void ParamSFOData::ValueData::SetData(const u8* data, int size)
 	u_size = size;
 }
 
+std::string ParamSFOData::generateFakeID(std::string filename)
+{
+	// Generates fake gameID for homebrew based on it's folder name.
+	// Should probably not be a part of ParamSFO, but it'll be called in same places.
+	std::string file = PSP_CoreParameter().fileToStart;
+	if (filename != "")
+		file = filename;
+
+	std::size_t lslash = file.find_last_of("/");
+	file = file.substr(lslash + 1);
+
+	int letter, sumOfAllLetters = 0;
+	for (int i = 0; i < file.size(); i++) {
+		letter = file[i];
+		sumOfAllLetters += letter;
+	}
+	
+	size_t j;
```

Comment:
Here also, `j` is not used outside.  Although this might be clearer:

```c++
int sumOfAllLetters = 0;
for (char &c : file) {
    c = toupper(c);
    sumOfAllLetters += c;
}
```

-[Unknown]